src/lib.rs

```diff
@@ -150,6 +170,76 @@ pub mod error;
 mod session;
 use crate::session::{Cmd, Session};
 
+
+
+/// Type alias for a Client with a Rustls connector
+///
+#[cfg(feature = "rustls-tls")]
+pub type RustlsClient = Client<hyper_rustls::HttpsConnector<hyper::client::HttpConnector>>;
+
+/// Type alias for a Client with an OpenSSL connector
+///
+#[cfg(feature = "openssl-tls")]
+pub type OpenSslClient = Client<hyper_tls::HttpsConnector<hyper::client::HttpConnector>>;
+
+pub use session::Client;
+
+/// Builder pattern for Client
+#[derive(Default, Clone, Debug)]
+pub struct ClientBuilder<C>
+where
+    C: connect::Connect + Send + Sync + Clone + Unpin
+{
+    capabalities: Option<webdriver::capabilities::Capabilities>,
+    connector: C,
+}
+
+#[cfg(feature = "rustls-tls")]
+impl ClientBuilder<hyper_rustls::HttpsConnector<hyper::client::HttpConnector>> {
+    /// Create a builder with Rustls connector, available with feature `rustls-tls`
+    pub fn rustls() -> Self {
+        Self::new(hyper_rustls::HttpsConnector::new())
+    }
+}
+
+#[cfg(feature = "openssl-tls")]
+impl ClientBuilder<hyper_tls::HttpsConnector<hyper::client::HttpConnector>> {
+    /// Create a builder with OpenSSL connector, available with feature `openssl-tls`
+    pub fn openssl() -> Self {
+        Self::new(hyper_tls::HttpsConnector::new())
+    }
+}
+impl<C> ClientBuilder<C>
+where
+    C: connect::Connect + Send + Sync + Clone + Unpin + 'static
+{
+
+    /// Create a new builder with TLS connector
+    /// # Arguments
+    /// * `connector` - TLS connector for [`Client`][hyper::client::Client]
+    pub fn new(connector: C) -> Self {
+        Self {
+            capabalities: None,
+            connector,
+        }
+    }
+    /// Set chosen webdriver capabilities
+    pub fn capabilities(mut self, cap: webdriver::capabilities::Capabilities) -> Self {
+        self.capabalities = Some(cap);
+        self
+    }
+    /// Connect to the webdriver session
+    /// # Arguments
+    /// * `webdriver` - webdriver url
+    pub async fn connect(self, webdriver: &str) -> Result<Client<C>, error::NewSessionError> {
+        return if let Some(cap) = self.capabalities {
```

Comment:
You don't need the `return` here -- in Rust the last statement is automatically returned.